packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -465,6 +465,13 @@ export class GarbageCollector implements IGarbageCollector {
          * For existing containers, we get this information from the metadata blob of its summary.
          */
         if (createParams.existing) {
+            // Fail fast if this container was created under a persisted test mode,
+            // since we don't know how it's implemented or how it should be used
+            const testMode = metadata?.gcTestMode;
+            if (testMode !== undefined) {
+                unreachableCase(testMode, `Cannot open container created under unknown GC Test Mode [${testMode}]`);
```

Comment:
I'm concerned this will cause document corruption

Reply:
That is the point

Reply:
Older clients will still be able to open this doc, it's ok if they succeed?

Reply:
I'd rather us have a soft enforcement rather than hard enforcement here.

Reply:
Suppose we find that a test mode itself is corrupting documents. We wouldn't want an unenlightened client opening it. 

Or imagine an old client collaborating with a SweepV0 client. It will be stuck with hardcoded Buffer so total
Sweep timeout will be over a day. Meanwhile if the SweepV0 client is summarizing/GCing, then stuff will start getting deleted early. Might work out, but might lead to weird bugs. It's just unsupported.

@vladsud am I seeing this right, or overengineering this? You had an example of a test mode leaking to production (blob aggregation or something?) would those learnings help think about preventative measures here?